01-node-tutorial/answers/writeWithPromisesThen.js

```diff
@@ -0,0 +1,23 @@
+const { writeFile, readFile } = require("fs").promises;
+
+writeFile("./temporary/temp.txt", "Line 1\n")
+  .then(() => {
+    return writeFile("./temporary/temp.txt", "Line 2\n", { flag: "a" });
+  })
+
+  .then(() => {
+    return writeFile("./temporary/temp.txt", "Line 3\n", { flag: "a" });
+  })
+
+  .then(() => {
+    return readFile("./temporary/temp.txt", "utf8");
+  })
+  .then((data) => {
+    console.log("Read content\n", data);
+  })
+
+  .catch((err) => {
+    console.log(err);
+  });
+  
+console.log("exited");
```

Comment:
Do you know why 'exited' is logged out before any other log in this code? Because, I think what you intended to do here is not what the result it.

If you don't, feel free to reachout via slack